lib/util.js

```diff
@@ -222,6 +222,44 @@ function urlToPath(path) {
     };
 }
 
+/**
+ * URL encodes each part of a URL path individually in order to support special
+ * characters # and &. Only http/https URLs are supported. Ensures that
+ * encoding happens only once, first by decoding and then re-encoding path parts.
+ *
+ * @param {String} path absolute URI whose path will be URL encoded. Supplied path
+ *  must be an absolute URL with protocal and host. For example:
+ *  https://host/path/to-be/encoded/myAsset.jpg
+ * @returns {String} absolute URI whose path has been URL encoded.
+ */
+function encodeUrlPath(path, base) {
+
+    const url = new URL(path, base);
+
+    if (url.protocol !== 'http:' && url.protocol !== 'https:' ) {
+        throw new UnsupportedFileUploadError(`only http/https urls are supported; url: ${path}`);
```

Comment:
```suggestion
        throw new UnsupportedFileUploadError(`only http/https urls are supported for encoding; url: ${path}`);
```
(clearer error message)